lib/modules/modal.js

```diff
@@ -36,20 +36,28 @@ export class _Modal extends React.Component {
     onToggleModal: () => {},
   };
 
+  _stopPropagation(e) {
+    e.stopPropagation();
+  }
+
+  _onClick = () => {
+    this.props.onToggleModal(null);
+  }
+
   render() {
-    const { className, show, children, onToggleModal } = this.props;
-
-    if (show) {
-      return (
-        <div style={ STYLE_CONTAINER } onClick={ () => onToggleModal(null) }>
-          <div className={ className } onClick={ e => e.stopPropagation() }>
```

Comment:
leaving a comment to mention that we need to further discuss this rule before this patch goes out.